//
// Calling non-functions raises an error.
//

try {
	was_error = false
	123()
} catch { was_error = true; }
assert(was_error, 'Invoking a number as function raises error')

try {
	was_error = false
	'something'()
} catch { was_error = true; }
assert(was_error, 'Invoking a string as function raises error')

try {
	was_error = false
	true()
} catch { was_error = true; }
assert(was_error, 'Invoking a bool as function raises error')

try {
	was_error = false
	null()
} catch { was_error = true; }
assert(was_error, 'Invoking a null as function raises error')

try {
	was_error = false
	[]()
} catch { was_error = true; }
assert(was_error, 'Invoking an array as function raises error')

try {
	was_error = false
	rx"\d+"()
} catch { was_error = true; }
assert(was_error, 'Invoking a regex as function raises error')
